android/java/AndroidManifest.xml

```diff
@@ -96,6 +96,14 @@
     android:screenOrientation="sensorPortrait"
     tools:ignore="LockedOrientationActivity,DiscouragedApi"/>
 
+<activity
+    android:name="org.chromium.chrome.browser.vpn.activities.VpnAlwaysOnActivity"
+    android:theme="@style/Theme.Chromium.Activity"
+    android:noHistory="true"
+    android:launchMode="singleTask"
+    android:screenOrientation="sensorPortrait"
+    tools:ignore="LockedOrientationActivity,DiscouragedApi"/>
```

Comment:
what API in particular triggers the `DiscouragedApi` warning?

Reply:
@SergeyZhukovsky it's mainly because of the portrait lock here `android:screenOrientation="sensorPortrait"`